src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/validator/PageRequestValidator.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.validator
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.Code
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.FieldError
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.ValidationError
+
+@Component
+class PageRequestValidator {
+  fun validatePageRequest(pageNumber: Int, pageSize: Int, maxAllowablePage: Int) {
+    val errors = mutableListOf<FieldError>()
+    if (pageNumber < 0) {
```

Comment:
nit: do we need CANNOT_BE_NEGATIVE _and_ CANNOT_BE_NEGATIVE_OR_ZERO ?

Reply:
well for `CANNOT_BE_NEGATIVE`, if the value is `0` and it's not allowed then it would be confusing to the caller.

Similarly for `CANNOT_BE_NEGATIVE_OR_ZERO`.

The only alternative is to provide `INVALID_VALUE` instead.